spec/usage/components/link_spec.rb

```diff
@@ -179,4 +179,140 @@ def response
     expect(stripped(static_output)).to ( include(stripped(expected_static_output)) )
   end
 
+  it "behaves correctly with anchor links (no reload, retain anchor)" do
+    class ExamplePage < Matestack::Ui::Page
+      def response
+        components {
+          link path: "#someanchor", text: "go to anchor", id: "my-link"
+
+          br times: 200
+
+          div id: "someanchor" do
+            plain "hello!"
+          end
+
+          div id: "my-div" do
+            plain "#{DateTime.now.strftime('%Q')}"
+          end
+        }
+      end
+    end
+
+    visit "/example"
+
+    element = page.find("#my-div")
+    before_content = element.text
+
+    # don't you rerender on me!
+    expect(ExamplePage).not_to receive(:new)
+
+    page.click_link("my-link")
+
+    # if the page reloaded we'd have different content here but as we don't want reloads
+    # we want the same
+    expect(page).to have_css("#my-div", text: before_content)
+    expect(page.current_url).to end_with("#someanchor")
+  end
+
+  describe "with an App" do
+
+    after :each do
+      class ExamplePage < Matestack::Ui::Page
+        # as the test suites work with redefining classes and this
+        # hack here was used to set a specific app easily,
+        # we need to "restore" previous state
+        def set_app_class
+          super
+        end
+      end
+    end
+
+    it "behaves correctly with anchor links (no reload, retain anchor) even inside an app" do
+      class Apps::MyTestApp < Matestack::Ui::App
+        def response
+          components {
+            page_content
+          }
+        end
+      end
+
+      class ExamplePage < Matestack::Ui::Page
+        def response
+          components {
+            link path: "#someanchor", text: "go to anchor", id: "my-link"
+
+            br times: 200
+
+            div id: "someanchor" do
+              plain "hello!"
+            end
+
+            div id: "my-div" do
+              plain "#{DateTime.now.strftime('%Q')}"
+            end
+          }
+        end
+
+        # Hacky/instable but easy way to set my custom App for this page
+        def set_app_class
+          @app_class = Apps::MyTestApp
```

Comment:
as the comment states, this is hacky but considering doing naming convetions, a new controller and appending to the routes etc. this seemed preferable to me.